src/shell/pipe_exec/command_not_found.rs

```diff
@@ -0,0 +1,35 @@
+use super::super::{Capture, Function, Shell};
+use parser::shell_expand::expand_string;
+use std::io::Read;
+use std::process;
+use sys;
+
+pub(crate) fn command_not_found(shell: &mut Shell, command: &str) -> Option<String> {
+    let function = match shell.functions.get("COMMAND_NOT_FOUND") {
+        Some(func) => func as *const Function,
+        None => return None // TODO: Use ? on Option whenever we drop support for older rust versions
```

Comment:
You can go ahead and use this feature now. Ion can only be built from nightly versions of Rust.

Reply:
(It can also be built by stable currently)
Ah yes! Just wanted to make sure, but forgot to make a comment on it in my self-review

Reply:
We use a number of features that are only available on nightly, and this project actually ensures that cargo selects a nightly version of Rust when compiling, even if your default toolchain is set to the latest stable.

Reply:
Wow, that is pretty epic! Didn't know that was possible!